src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -434,4 +482,32 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
     {
         onMobxPromise(this.transposedCnaDataText, text=>fileDownload(text, "cna_transposed.txt"));
     }
+
+    @action
+    private handleQueryButtonClick(querySampleIds: string) {
+        this.props.store.selectedSampleIdsForQueryButton = querySampleIds;
+        this.props.store.queryFormVisible = true;
+    }
+
+    @action
+    private async handleVisualStudyButtonClick(alterations: ICaseAlteration[]) {
+        let studies = _.reduce(_.groupBy(alterations, "studyId"), (acc: { id: string; samples: string[] }[], alteration, studyId) => {
+            acc.push({
+                id: studyId,
+                samples: alterations.map(alteration => alteration.sampleId)
+            })
+            return acc;
+        }, []);
+
+        let parameters = {
+            name: "visual study",
+            description: "visual study",
+            origin: studies.map(study => study.id),
+            studies: studies
+        }
+        const saveStudyResult = await sessionServiceClient.saveVirtualStudy(parameters, false);
+        if (saveStudyResult && saveStudyResult.id) {
+            window.open(buildCBioPortalPageUrl({pathname:'study', query: {id: saveStudyResult.id}}), "_blank");
```

Comment:
@alisman Just talked with @jjgao in person. Rather than create a virtual study, we can just open a dropdown as same as the one in study view page.
![image](https://user-images.githubusercontent.com/15748980/59532468-d4fc3000-8eb6-11e9-99da-ac25e8f93243.png)
So we can let the user decide to save/share the virtual study or just open it.

Reply:
cool!